lib/libzfs/libzfs_import.c

```diff
@@ -173,8 +175,23 @@ fix_paths(nvlist_t *nv, name_entry_t *names)
 				break;
 			}
 
-			if (best == NULL || ne->ne_order < best->ne_order)
+			if (best == NULL) {
```

Comment:
There are definitely a few ways to structure this.  After rewriting this 3 times I opted for the `if-continue` blocks which I think are more readable because they allow a comment to be clearly placed above each block.  Putting everything in an `if / else if / else if` or `if (a || b || c)` block probably means I'd need to write a single big comment above the whole thing explaining all the case.  Which would be fine too, although I agree that last continue is a little odd.